Shared/View/VoteView.swift

```diff
@@ -8,28 +8,50 @@
 import SwiftUI
 
 struct VoteView: View {
-    // Hardcoded values to test
-    @State private var tempActivty = Activity(
-        id: "123",
-        title: "What movie should we watch tonight?",
-        category: Category.movie,
-        // FIXME: this is not a valid user ID
-        // TODO: request renaming by backend if this is confusing
-        author: "mightyDonut"
-    )
+    @EnvironmentObject var user: UserViewModel
+    @EnvironmentObject var activity: ActivityViewModel
+    @EnvironmentObject var option: OptionViewModel
+
     @State private var optionArray = ["Shrek", "Wall-E", "ET"]
     @State private var isSelected = ""
     @State private var newOption = ""
+        
+    // Some default values, will automatically be replaced when call to .activity() method is made
+    @State private var act = Activity(id: "1", title: "Movie Night", category: Category.movie, author: "1")
+    // VoteView retrieves title, category by using activity(id: Activity.ID) from ActivityViewModel
     var body: some View {
         VStack(alignment: .leading) {
-            Text("Activity").font(.title).padding(.bottom)
-            Text(tempActivty.title).font(.title3)
-            Text("Category: \(tempActivty.category.rawValue.capitalized)").foregroundColor(.secondary)
+            // Get an activity to retrieve data (title, category, options)
+            // Not sure how to resolve Static method 'buildBlock' requires
+            // that 'Task<(), Never>' conform to 'View' without using Button...
+            Button {
+                Task {
+                    do {
+                        act = try await ActivityViewModel.activity(id: "AhdjSLOlTJqb68aXBAWn")
+                    } catch {
+                        print(error)
+                    }
+                }
+            } label: {
+                NavigationLink(destination: VoteView()) {
+                    Text("Get Activity")
+                }
+            }
+            
+            // Testing purposes
+            // Text("Activity \(act.title)")
+            // Text("Activity").font(.title).padding(.bottom)
+            
+            // Testing purposes for retrieving the corresponding activityID?
+            // Text("activityid \(activity.activity?.id ?? "")")
+            
+            Text("Activity \(act.title)").font(.title3)
+            Text("Category: \(act.category.rawValue.capitalized)").foregroundColor(.secondary)
             Text("Options").font(.title).padding(.top)
             // Option selection
             ScrollView {
                 VStack(alignment: .leading) {
-                    // TODO: Correctly update fields from Model files when user inputs data
+                    // TODO: Loop through the activity options array...
```

Comment:
As you might have noticed, that's not an array, but rather a dictionary (or Map in Java). My recommendation is to make the keys of the dictionary an array to ForEach on, so something like

```swift
// somewhere else in the struct where you define the other variables:
var optionIDs: [Option.ID] {
   act.options.keys.sorted() // can sort in some other ways
}

// inside the body:
ForEach(optionIDs, id: \.self) { optionID in
   ...
}
```